ext/pdclient/OAIDefaultApi.cpp

```diff
@@ -0,0 +1,1175 @@
+    QString error_str = worker->error_str;
+    QNetworkReply::NetworkError error_type = worker->error_type;
+
+    if (worker->error_type != QNetworkReply::NoError) {
+        error_str = QString("%1, %2").arg(worker->error_str, QString(worker->response));
+    }
+    worker->deleteLater();
+
+    if (worker->error_type == QNetworkReply::NoError) {
+        emit unloadingCompleteActionIdPostSignal();
+        emit unloadingCompleteActionIdPostSignalFull(worker);
+    } else {
+        emit unloadingCompleteActionIdPostSignalE(error_type, error_str);
+        emit unloadingCompleteActionIdPostSignalEFull(worker, error_type, error_str);
+    }
+}
+
+void OAIDefaultApi::unloadingPendingGet() {
+    QString fullPath = QString(_serverConfigs["unloadingPendingGet"][_serverIndices.value("unloadingPendingGet")].URL()+"/unloading/pending");
+    
+    OAIHttpRequestWorker *worker = new OAIHttpRequestWorker(this, _manager);
+    worker->setTimeOut(_timeOut);
+    worker->setWorkingDirectory(_workingDirectory);
+    OAIHttpRequestInput input(fullPath, "GET");
+
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
+    for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
+        input.headers.insert(keyValueIt->first, keyValueIt->second);
+    }
+#else
+    for (auto key : _defaultHeaders.keys()) {
+        input.headers.insert(key, _defaultHeaders[key]);
+    }
+#endif
+
+    connect(worker, &OAIHttpRequestWorker::on_execution_finished, this, &OAIDefaultApi::unloadingPendingGetCallback);
+    connect(this, &OAIDefaultApi::abortRequestsSignal, worker, &QObject::deleteLater);
+    connect(worker, &QObject::destroyed, this, [this]() {
+        if (findChildren<OAIHttpRequestWorker*>().count() == 0) {
+            emit allPendingRequestsCompleted();
+        }
+    });
+
+    worker->execute(&input);
+}
+
+void OAIDefaultApi::unloadingPendingGetCallback(OAIHttpRequestWorker *worker) {
+    QString error_str = worker->error_str;
+    QNetworkReply::NetworkError error_type = worker->error_type;
+
+    if (worker->error_type != QNetworkReply::NoError) {
+        error_str = QString("%1, %2").arg(worker->error_str, QString(worker->response));
+    }
+    OAIActionStatusList output(QString(worker->response));
+    worker->deleteLater();
+
+    if (worker->error_type == QNetworkReply::NoError) {
+        emit unloadingPendingGetSignal(output);
+        emit unloadingPendingGetSignalFull(worker, output);
+    } else {
+        emit unloadingPendingGetSignalE(output, error_type, error_str);
+        emit unloadingPendingGetSignalEFull(worker, error_type, error_str);
+    }
+}
+
+void OAIDefaultApi::unloadingPost(const OAIContainerRequest &oai_container_request) {
+    QString fullPath = QString(_serverConfigs["unloadingPost"][_serverIndices.value("unloadingPost")].URL()+"/unloading");
+    
+    OAIHttpRequestWorker *worker = new OAIHttpRequestWorker(this, _manager);
```

Comment:
delete the pointer after finish using it

Reply:
This is generated code it does not need to be edited